### PR TITLE
hostmetricsreceiver: refactor load metrics to use generated metrics

### PR DESCRIPTION
**Pull request by mstumpfx**
**Description:** Adding the load metrics to the metadata.yaml, and refactoring the hostmetricreceivers load metrics to use generated metrics

### CODE REVIEW

File: cmd/mdatagen/metrics.tmpl [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 // Type is the component type name.
 const Type configmodels.Type = "{{ .Name }}"
 
-type metricIntf interface {
+type MetricIntf interface {
```
Comment (jrcamp) [inline cmd/mdatagen/metrics.tmpl:27]: Can we just call it Metric?

**Review by jrcamp — COMMENTED** [review]
lgtm

File: cmd/mdatagen/metrics.tmpl (hunk 1 above)
Comment (mstumpfx) [reply cmd/mdatagen/metrics.tmpl]: Sure - updated.

**Review by jrcamp — APPROVED** [review]

File: receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
 // Type is the component type name.
 const Type configmodels.Type = "hostmetricsreceiver"
 
-type metricIntf interface {
+type Metric interface {
```
Comment (tigrannajaryan) [inline receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go:27]: 1. Does this need to be exported? (It was not previously).
2. Is it possible to come up with a less generic name? Looking at the name `Metric` I have a hard time understanding what it is supposed to do.

Comment (mstumpfx) [reply receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go:27]: @tigrannajaryan I had changed it to exported to be able to pass through the interface so the Init() function could be called generically - https://github.com/open-telemetry/opentelemetry-collector/blob/db35667fe72095b0585e1ca4edc9767f4ddbcdbe/receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper.go#L69-L76 
I think pattern will occur more as more of the metrics in the receiver are refactored to be based off the generated metrics. The previous use cases were only for the scrapers that emitted a single metric.

I'm indifferent towards the name - `MetricIntf`? `MetricFactory`? `GeneratedMetric`?

Comment (tigrannajaryan) [reply receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go:27]: OK, understood re exporting.

As for the name, please use something that reflects the purpose. I do not quite understand what it does to suggest a good name. The generic name `Metric` does not help me to understand what it is doing.

If unsure keep it unchanged.

Comment (mstumpfx) [reply receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go]: I reverted the name back to `MetricIntf` for now and added a comment above to describe it.

**Pull request merged**